plasmapy/particles/_parsing.py

```diff
@@ -477,34 +477,33 @@ def parse_and_check_molecule_input(argument: str, Z: Integral = None):
 
     Parameters
     ----------
-    argument : 'str'
+    argument : `str`
         The molecule symbol to be parsed.
 
-    Z : 'Integral', optional
+    Z : integer, optional
```

Comment:
```suggestion
    Z : `int`, optional
```

Why not this?  I'm sure you purposefully used integer over int, but I'm missing the reason.

Reply:
Use `numpy.integer` then.  If I remember correctly, that represents all integer dtypes and the Python integer.

Reply:
I was wrong.  It only represents all numpy integer dtypes.  You'd need to do something like `Union[int, numpy.integer]`.

Reply:
```Python
>>> isinstance(1, np.integer)
False
```
Hm...doesn't include the Python integer, unfortunately...

Reply:
Yep.... :( It would have to be this... I always forget this...

```Python
>>> isinstance(5, (int, numpy.integer))
True
>>> isinstance(numpy.int32(5), (int, numpy.integer))
True
```

Reply:
I'll go with `int` for now since that's our standard practice.  I'm wondering if we could do a ``` :term:`integer` ``` here and link to a glossary entry, but I'm pretending to have learned my lesson about scope creep.